### ISSUE TITLE
No case for ToDoubleNodeGen node with values of type BigDecimal

### ISSUE THREAD

**Issue by gogainda** [oracle/truffleruby#2815]
Found [here](https://github.com/gogainda/lockbox/blob/master/test/model_types_test.rb#L195)
```
ruby -e "require 'bigdecimal'; puts [BigDecimal('123456789.123456789123456789')].pack('G')"


-e:1:in `pack': TruffleRuby doesn't have a case for the org.truffleruby.core.format.convert.ToDoubleNodeGen node with values of type BigDecimal(org.truffleruby.core.basicobject.RubyBasicObject) (TypeError)
	from org.truffleruby.core.format.convert.ToDoubleNodeGen.executeAndSpecialize(ToDoubleNodeGen.java:136)
	from org.truffleruby.core.format.convert.ToDoubleNodeGen.executeToDouble(ToDoubleNodeGen.java:55)
	from org.truffleruby.core.format.read.array.ReadDoubleNode.read(ReadDoubleNode.java:33)
	from org.truffleruby.core.format.read.array.ReadDoubleNodeGen.executeAndSpecialize(ReadDoubleNodeGen.java:100)
	from org.truffleruby.core.format.read.array.ReadDoubleNodeGen.execute(ReadDoubleNodeGen.java:64)
	from org.truffleruby.core.format.convert.ReinterpretAsLongNodeGen.execute(ReinterpretAsLongNodeGen.java:27)
	from org.truffleruby.core.format.convert.Integer64BigToBytesNodeGen.execute(Integer64BigToBytesNodeGen.java:27)
	from org.truffleruby.core.format.write.bytes.WriteBytesNodeGen.execute(WriteBytesNodeGen.java:31)
	from org.truffleruby.core.format.FormatRootNode.execute(FormatRootNode.java:57)
	from -e:1:in `<main>'
```

**Comment by andrykonchin** [comment]
Thank you for reporting!

**Comment by andrykonchin** [comment]
Fixed in 078b9721bb4dbd971d05f3eabbffd73fafde3c35
